### PR TITLE
[ci] Fetch submodules and always update cache.

### PR DESCRIPTION
**Pull request by stellaraccident**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
       - name: Install deps
         run: |
           apt update
-          apt install libboost-all-dev
+          apt install -y libboost-all-dev
```
Comment (makslevental) [inline .github/workflows/ci.yml:47]: just a btw: there's a [marketplace action for a this](https://github.com/Xilinx/mlir-air/blob/wheels_1/.github/workflows/distro.yml#L102-L108) that's more "portable".

Comment (stellaraccident) [reply .github/workflows/ci.yml:49]: I am going to spend no time on this. Boost must go.

Comment (stellaraccident) [reply .github/workflows/ci.yml:49]: (less snark: I'll just add it to the docker image that is behind this)

Comment (stellaraccident) [reply .github/workflows/ci.yml]: Reverted this change (it is in the GH actions runner image now)

**Pull request merged**